File: C++/Level.1/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int clothes[n+1];
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    fill(clothes, clothes + n + 1, 1);
    
    for(int i : lost) {
        --clothes[i];
    }
    
    for(int i : reserve) {
        ++clothes[i];
    }
    
    for(int i = 1; i <= n; ++i) {
        if(clothes[i] > 1) {
            if(clothes[i-1] == 0) {
                --clothes[i];
                ++clothes[i-1];
                continue;
            }
            if(i != n && clothes[i+1] == 0) {
                --clothes[i];
                ++clothes[i+1];
                continue;
            }
        }
    }
    
    for(int i = 1; i <= n; ++i) {
        if(clothes[i] > 0) ++answer;
    }
    
    return answer;
}

File: JavaScript/Level.1/체육복.js
function solution(n, lost, reserve) {
    let answer = 0;
    let student = Array.from({ length: n + 1 }, () => 1);

    lost.forEach((e) => {
        --student[e];
    });
    reserve.forEach((e) => {
        ++student[e];
    });
    for (let i = 1; i <= n; ++i) {
        if (student[i] === 0) {
            if (i > 1 && student[i - 1] > 1) {
                --student[i - 1];
                ++student[i];
            } else if (i < n && student[i + 1] > 1) {
                --student[i + 1];
                ++student[i];
            }
        }
        if (student[i] > 0) {
            ++answer;
        }
    }
    return answer;
}
